connection.go

```diff
@@ -403,18 +412,20 @@ func (c *connection) Fetch(req *proto.FetchReq) (*proto.FetchResp, error) {
 				}
 				partition.Messages = partition.Messages[i:]
 			} else {
-				firstOffset := partition.RecordBatch.FirstOffset
-				for _, rec := range partition.RecordBatch.Records {
-					if firstOffset+rec.OffsetDelta >= requestedOffset {
-						break
+				for _, rb := range partition.RecordBatches {
+					i := 0
```

Comment:
@e-max please look at this part carefully.  It was not tested before, and it definitely got trickier once we switched to multiple record batches in a response.  I'm pretty sure I broke it temporarily before adding the test.

It's not clear to me how to trigger this from a real server to make sure that my test case is even correct.

It seems plausible that we'd end up with an empty batch.  In that case, should we actually remove the batch itself?

Also, is it safe to stop after we find an offset higher than the requested offset?  If so, it's a (small) waste of time to keep checking the rest of the batches.

Reply:
> It seems plausible that we'd end up with an empty batch. In that case, should we actually remove the batch itself?
I wouldn't bother.  Because we anyway repack all these messages in our structure []Messages.


> Also, is it safe to stop after we find an offset higher than the requested offset? If so, it's a (small) waste of time to keep checking the rest of the batches. 
I think it's safe. It's the last thing we read from reader, and we don't reuse buffers, so there is no danger of memory leaks.

Reply:
I think I'll leave it like this for now anyway.  Doesn't seem worth the added complication.